### PR TITLE
test: fix compile errors in `commands.ts`

### PR DESCRIPTION
**Pull request by jonas-jonas**
Fixes a few compile time issues in the cypress commands and adds generation of a type declaration for the Kratos configuration to be used within E2E tests.

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: test/e2e/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "cypress": "^11.2.0",
     "dayjs": "^1.10.4",
     "got": "^11.8.2",
+    "json-schema-to-typescript": "^12.0.0",
```
Comment (jonas-jonas) [inline test/e2e/package.json:19]: This tool is pretty cool! We now have a generated `d.ts` file of our config, that we can use in the tests to get typesafe config updates. Always bugged me, that I wasn't able to get code completion when working with it.

https://github.com/bcherny/json-schema-to-typescript#readme

File: test/e2e/cypress/integration/profiles/webhoooks/registration/success.spec.ts [hunk 2]
```diff
@@ -147,7 +147,9 @@ context("Registration success with email profile with webhooks", () => {
           expect(identity.schema_id).to.equal("default")
           expect(identity.schema_url).to.equal(`${APP_URL}/schemas/ZGVmYXVsdA`)
           expect(identity.traits.email).to.equal("updated-" + email)
-          expect(identity.metadata_public.some).to.equal("public fields")
+          expect((identity as any).metadata_public.some).to.equal(
```
Comment (jonas-jonas) [inline test/e2e/cypress/integration/profiles/webhoooks/registration/success.spec.ts:150]: These types still use a pretty old version of `@ory/kratos-client`, where `metadata_public` didn't exist, yet. But I want to do that in a future PR.

File: test/e2e/cypress/support/config.d.ts [hunk 3]
```diff
@@ -0,0 +1,1185 @@
...
+ * This file was automatically generated by json-schema-to-typescript.
+ * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
+ * and run json-schema-to-typescript to regenerate this file.
+ */
+
+export type OryKratosConfiguration = OryKratosConfiguration1 &
+  OryKratosConfiguration2
+export type OryKratosConfiguration1 = {
+  [k: string]: unknown | undefined
+}
```
Comment (jonas-jonas) [inline test/e2e/cypress/support/config.d.ts:15]: Not sure, why it generates these 1, 2 versions, but I traced it to being related to the `allOf` block at the end of the config. Not sure if there is anything we can do about that here.

**Review by aeneasr — APPROVED** [review]

File: test/e2e/package.json (hunk 1 above)
Comment (CaptainStandby) [reply test/e2e/package.json:19]: cool idea

**Pull request merged**